crates/yas_periphery/src/yas_nft_position_manager.cairo

```diff
@@ -1,24 +1,635 @@
+
+            // compute the liquidity amount
+            let (sqrt_price_X96, _, _, _) = pool_dispatcher.slot_0();
+            let sqrt_ratio_AX96 = get_sqrt_ratio_at_tick(params.tick_lower);
+            let sqrt_ratio_BX96 = get_sqrt_ratio_at_tick(params.tick_upper);
+
+            let liquidity = get_liquidity_for_amounts(
+                sqrt_price_X96,
+                sqrt_ratio_AX96,
+                sqrt_ratio_BX96,
+                params.amount_0_desired,
+                params.amount_1_desired
+            );
+
+            let (amount_0, amount_1) = pool_dispatcher
+                .mint(
+                    params.recipient,
+                    params.tick_lower,
+                    params.tick_upper,
+                    liquidity,
+                    array![get_caller_address().into()]
+                );
+
+            assert(
+                amount_0 >= params.amount_0_min && amount_1 >= params.amount_1_min,
+                'Price slippage check'
+            );
+            (liquidity, amount_0, amount_1, pool_dispatcher)
+        }
+    }
+
+    /// @notice Computes the amount of liquidity received for a given amount of token0 and price range
+    /// @dev Calculates amount0 * (sqrt(upper) * sqrt(lower)) / (sqrt(upper) - sqrt(lower))
+    /// @param sqrtRatioAX96 A sqrt price representing the first tick boundary
+    /// @param sqrtRatioBX96 A sqrt price representing the second tick boundary
+    /// @param amount0 The amount0 being sent in
+    /// @return liquidity The amount of returned liquidity
+    fn get_liquidity_for_amount_0(
+        sqrt_ratio_AX96: FixedType, sqrt_ratio_BX96: FixedType, amount_0: u256
+    ) -> u128 {
+        let (sqrt_ratio_AX96, sqrt_ratio_BX96) = if sqrt_ratio_AX96 > sqrt_ratio_BX96 {
+            (sqrt_ratio_BX96, sqrt_ratio_AX96)
+        } else {
+            (sqrt_ratio_AX96, sqrt_ratio_BX96)
+        };
+        // TODO: .mag
+        let intermediate = FullMath::mul_div(sqrt_ratio_AX96.mag, sqrt_ratio_BX96.mag, ONE);
+        FullMath::mul_div(amount_0, intermediate, (sqrt_ratio_BX96 - sqrt_ratio_AX96).mag)
+            .try_into()
+            .unwrap()
+    }
+
+    /// @notice Computes the amount of liquidity received for a given amount of token1 and price range
+    /// @dev Calculates amount1 / (sqrt(upper) - sqrt(lower)).
+    /// @param sqrtRatioAX96 A sqrt price representing the first tick boundary
+    /// @param sqrtRatioBX96 A sqrt price representing the second tick boundary
+    /// @param amount1 The amount1 being sent in
+    /// @return liquidity The amount of returned liquidity
+    fn get_liquidity_for_amount_1(
+        sqrt_ratio_AX96: FixedType, sqrt_ratio_BX96: FixedType, amount_1: u256
+    ) -> u128 {
+        let (sqrt_ratio_AX96, sqrt_ratio_BX96) = if sqrt_ratio_AX96 > sqrt_ratio_BX96 {
+            (sqrt_ratio_BX96, sqrt_ratio_AX96)
+        } else {
+            (sqrt_ratio_AX96, sqrt_ratio_BX96)
+        };
+        // TODO: .mag
```

Comment:
What do you mean you want to return the `mag` of this result? There is no need for that in this case because you are already working with the `mag` right?

Reply:
✔️